src/metabase/models/database.clj

```diff
@@ -49,9 +49,12 @@
     (schedule-tasks! database)))
 
 (defn- post-select [{driver :engine, :as database}]
-  ;; TODO - this is only really needed for API responses. This should be a `hydrate` thing instead!
   (cond-> database
-    (driver/initialized? driver) (assoc :features (driver.u/features driver))))
+    (driver/initialized? driver)
+    ;; TODO - this is only really needed for API responses. This should be a `hydrate` thing instead!
+    (assoc :features (driver.u/features driver))
+    (get-method driver/normalize-db-details driver)
```

Comment:
@camsaul , not sure if this is what you meant, in terms of normalizing.

I put it in post-select since the hook already existed, and we want the UI to be populated with the updated values coming out of the DB (which presumably also overwrite the existing values on save, thereby removing the `:regionid` key).

It doesn't seem quite optimal, though, since this will run on **every** select.  I suppose the only way to prevent that would be to write a Clojure migration?

Reply:
Oh i see why you did it this way. I'd add a `:default` implementation that just returned the details rather than optionally calling this or not.

Reply:
Ah yeah, that's better.  Will make the change.

Reply:
Implementation note (for posterity): @camsaul  had suggested using the `:types` facility of Toucan do to this work.  However, I opted to add it here instead, because:

- `:details` is already type `:encrypted-json`, which has its own `:out` handling going on (to decrypt the JSON), so that would need to be changed to perform this additional step (and possibly change the semantics of that field type)
- this `post-select` fn was already in place, I'm just changing the initial value that is threaded through

Still, if there is a cleaner way, I'm open to it.